src/Debug.c

```diff
@@ -48,28 +48,28 @@ void Ros_Debug_BroadcastMsg(char* fmt, ...)
 
     if (ros_DebugSocket == -1)
         Ros_Debug_Init();
+
     // Timestamp
     //The timestamp for the message "Found Micro-Ros PC Agent" will be the epoch time (THU 1970-01-01 00:00:00.000) as the global flags 
     //are set to indicate that the Micro-Ros PC Agent is connected but the first sync of the host time using the micro-ROS agent is yet to occur
     struct tm synced_time;
-    timespec tp;
+    struct timespec tp;
     char timestamp[FORMATTED_TIME_SIZE];
-    builtin_interfaces__msg__Time debug_msg_timestamp;
     if (g_Ros_Communication_AgentIsConnected)
     {
         //get synchronized time from the agent
-        int64_t nanosecs = rmw_uros_epoch_nanos(); 
-        Ros_Nanos_To_Time_Msg(nanosecs, &debug_msg_timestamp);
-        strftime(timestamp, FORMATTED_TIME_SIZE, "%Y-%m-%d %H:%M:%S", localtime_r((const time_t*)&debug_msg_timestamp.sec, &synced_time));
-        snprintf(timestamp + strlen(timestamp), FORMATTED_TIME_SIZE - strlen(timestamp), ".%06d ", (int)debug_msg_timestamp.nanosec / 1000);
+        int64_t nanosecs = rmw_uros_epoch_nanos();
+        Ros_Nanos_To_Timespec(nanosecs, &tp);
     }
     else
     {
         //rmw_uros_epoch_nanos cannot sync with agent because it's not connected
         clock_gettime(CLOCK_REALTIME, &tp);
-        strftime(timestamp, FORMATTED_TIME_SIZE, "%Y-%m-%d %H:%M:%S", localtime_r(&tp.tv_sec, &synced_time));
-        snprintf(timestamp + strlen(timestamp), FORMATTED_TIME_SIZE - strlen(timestamp), ".%06d ", tp.tv_nsec / 1000);
     }
+    localtime_r(&tp.tv_sec, &synced_time);
```

Comment:
Using `localtime_r(..)` like this facilitates use on FS100 a bit as well, as we don't use the return value directly any more (which is different between FS100 and the rest).